lib/resources/mssql_session.rb

```diff
@@ -25,26 +25,27 @@ class MssqlSession < Inspec.resource(1)
 
       # Passing no credentials to mssql_session forces it to use Windows authentication
       sql_windows_auth = mssql_session
-      describe sql.query(\"SELECT SERVERPROPERTY('IsIntegratedSecurityOnly') as \\\"login_mode\\\";\").row(0).column('login_mode') do
+      describe sql_windows_auth.query(\"SELECT SERVERPROPERTY('IsIntegratedSecurityOnly') as \\\"login_mode\\\";\").row(0).column('login_mode') do
         its('value') { should_not be_empty }
         its('value') { should cmp == 1 }
       end
     "
 
-    attr_reader :user, :password, :host
+    attr_reader :user, :password, :host, :port, :instance
     def initialize(opts = {})
       @user = opts[:user]
       @password = opts[:password] || opts[:pass]
       if opts[:pass]
         warn '[DEPRECATED] use `password` option to supply password instead of `pass`'
       end
       @host = opts[:host] || 'localhost'
+      @port = opts[:port] || '1433'
       @instance = opts[:instance]
 
       # check if sqlcmd is available
-      return skip_resource('sqlcmd is missing') if !inspec.command('sqlcmd').exist?
+      raise Inspec::Exceptions::ResourceFailed, 'sqlcmd is missing' unless inspec.command('sqlcmd').exist?
```

Comment:
This is technically a breaking change since we mark the resource as "skipped" and not failed. In principle, I think I agree with you that we should fail, but we have to do that during a major version bump.

For now, we should change this to `Inspec::Exceptions::ResourceSkipped`